src/mantaray-v1.ts

```diff
@@ -0,0 +1,738 @@
+    const fork = this.forks[path[0]]
+
+    if (!fork) {
+      const newNode = new MantarayNode()
+
+      if (!equalBytes(this._obfuscationKey, null32Bytes)) {
+        if (!obfuscationKeyGenerator) {
+          throw new RandomBytesFnUndefined()
+        }
+        newNode.obfuscationKey = obfuscationKeyGenerator()
+      }
+
+      // Continuous node
+      if (path.length > nodeForkSizes.prefixMax) {
+        const prefix = path.slice(0, nodeForkSizes.prefixMax)
+        const rest = path.slice(nodeForkSizes.prefixMax)
+        newNode.addFork(rest, attributes)
+        newNode.isContinuousNode = true
+        this.forks[path[0]] = new MantarayFork(prefix, newNode)
+        this._isEdge = true
+        this.makeDirty()
+
+        return
+      }
+
+      // create non-continuous node
+      if (entry) newNode.entry = entry
+
+      newNode.forkMetadata = forkMetadata
+      newNode.nodeMetadata = nodeMetadata
+
+      this.forks[path[0]] = new MantarayFork(path, newNode)
+      this.makeDirty()
+      this._isEdge = true
+
+      return
+    }
+
+    /// Case when there is an existing fork for the given (sub)path
+
+    const commonPath = common(fork.prefix, path)
+    /** restPath of the existing fork's path that differs from the new path */
+    const restPath = fork.prefix.slice(commonPath.length)
+    let newNode = fork.node
+
+    // create new parent node where the path starts to differ in the current node's prefix
+    // this parent node will have two children: the current node and the new node with the given path
+    if (restPath.length > 0) {
+      // create new node for the common path
+      newNode = new MantarayNode()
+
+      if (equalBytes(this._obfuscationKey, null32Bytes)) {
+        newNode.obfuscationKey = new Uint8Array(32) as Bytes<32>
+      } else {
+        if (!obfuscationKeyGenerator) {
+          throw new RandomBytesFnUndefined()
+        }
+        newNode.obfuscationKey = obfuscationKeyGenerator()
+      }
+      newNode.forks = {}
+      const newFork = new MantarayFork(restPath, fork.node)
+
+      if (fork.node.isContinuousNode) {
+        handleTrimmedContinuousFork(newFork)
+      }
+      newNode.forks[restPath[0]] = newFork // copy old parent node to its remaining path
+      newNode._isEdge = true
+    }
+
+    // NOTE: special case on edge split
+    // newNode will be the common path edge node
+    // newNode's prefix is a subset of the given `path`, here the desired fork will be added with the truncated path
+    newNode.addFork(path.slice(commonPath.length), attributes)
+    this.forks[path[0]] = new MantarayFork(commonPath, newNode)
+    this._isEdge = true
+
+    this.makeDirty()
+  }
+
+  /**
+   * Gives back a MantarayFork under the given path
+   *
+   * @param path valid path within the MantarayNode
+   * @returns MantarayFork with the last unique prefix and its node
+   * @throws error if there is no node under the given path
+   */
+  public getForkAtPath(path: Uint8Array): MantarayFork {
```

Comment:
well, it is a bit problematic because at string serialization by definition it has to allow only ASCII characters, but there could be other use-cases when the path is not text based or could work somehow with UTF8 char set as well (the latest I haven't tested and I wanted to keep now as it is like in the previous version). In a different PR/issue it may worth to investigate this side.